cli/azd/pkg/account/manager.go

```diff
@@ -225,39 +225,22 @@ func (m *Manager) getAllSubscriptions(ctx context.Context) ([]*azcli.AzCliSubscr
 func (m *Manager) getDefaultSubscription(ctx context.Context) (*Subscription, error) {
 	// Get the default subscription ID from azd configuration
 	configSubscriptionId, ok := m.config.Get(defaultSubscriptionKeyPath)
-	var defaultSubscription *Subscription
-
-	if ok {
-		subscriptionId := fmt.Sprint(configSubscriptionId)
-		subscription, err := m.azCli.GetAccount(ctx, subscriptionId)
-		if err != nil {
-			log.Printf("failed retrieving subscription with ID '%s'. %s", subscriptionId, err.Error())
-		}
-
-		defaultSubscription = &Subscription{
-			Id:       subscription.Id,
-			Name:     subscription.Name,
-			TenantId: subscription.TenantId,
-		}
-	} else {
-		// No defaults subscription has been set in azd config
-		allSubscriptions, err := m.getAllSubscriptions(ctx)
-		if err != nil {
-			return nil, fmt.Errorf("failed retrieving subscriptions for current account. %w", err)
-		}
 
-		if len(allSubscriptions) == 0 {
-			return nil, errors.New("no subscriptions found for current account")
-		}
+	if !ok {
+		return nil, nil
+	}
 
-		defaultSubscription = &Subscription{
-			Id:       allSubscriptions[0].Id,
-			Name:     allSubscriptions[0].Name,
-			TenantId: allSubscriptions[0].TenantId,
-		}
+	subscriptionId := fmt.Sprint(configSubscriptionId)
+	subscription, err := m.azCli.GetAccount(ctx, subscriptionId)
+	if err != nil {
+		return nil, fmt.Errorf("failed retrieving subscription with ID '%s'. %w", subscriptionId, err)
```

Comment:
We might be able to do more here.
This basically means that there's a value in `config.defaults` for the subscription and is not valid. So, we can ask the user if he wants to remove it (if running interactive).  Or we can just remove it from the config.default.
Or we can mention in the error message that the value is not valid or unknown and what to do to fix it (call azd config foo bar).

The current message might leave the user wondering `and what do I do now???`

Reply:
Added more verbose error message for now.